content/getting-started/java.textile

```diff
@@ -0,0 +1,246 @@
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.ClientOptions;
+import io.ably.lib.types.ErrorInfo;
+import io.ably.lib.types.Message;
+import io.ably.lib.types.PaginatedResult;
+
+public class App {
+    public static void main(String[] args) {
+        try {
+            // Initialize the Ably Realtime client
+            ClientOptions options = new ClientOptions("{{API_KEY}}");
+            options.clientId = "my-first-client";
+            AblyRealtime realtime = new AblyRealtime(options);
+
+            // Wait for the connection to be established
+            realtime.connection.on(ConnectionEvent.connected, connectionStateChange -> {
+                System.out.println("Made my first connection!");
+            });
+
+            // Keep the program running
+            Thread.sleep(2000);
+
+        } catch (AblyException e) {
+            e.printStackTrace();
+        }
+    }
+}
+```
+
+You can monitor the lifecycle of clients' connections by registering a listener that will emit an event every time the connection state changes. For now, run the program with @./gradlew run@ to log a message to the console to know that the connection attempt was successful. You'll see the message printed to your console, and you can also inspect the connection event in the dev console of your app.
+
+h2(#step-2). Step 2: Subscribe to a channel and publish a message
+
+Messages contain the data that a client is communicating, such as a short 'hello' from a colleague, or a financial update being broadcast to subscribers from a server. Ably uses channels to separate messages into different topics, so that clients only ever receive messages on the channels they are subscribed to.
+
+* Add the following lines to your @main@ method, above the line @// Keep the program running@, to create a channel instance and register a listener to subscribe to the channel. Then run it with @./gradlew run@:
+
+```[java]
+// Get a channel instance
+Channel channel = realtime.channels.get("my-first-channel");
+
+// Subscribe to messages on the channel
+channel.subscribe(message -> {
+    System.out.println("Received message: " + message.data);
+});
+```
+
+* Use the Ably CLI to publish a message to your first channel. The message will be received by the client you've subscribed to the channel, and be logged to the console.
+
+```[sh]
+ably channels publish my-first-channel 'Hello!' --name "myEvent"
+```
+
+* In a new terminal tab, subscribe to the same channel using the CLI:
+
+```[sh]
+ably channels subscribe my-first-channel
+```
+
+Publish another message using the CLI and you will see that it's received instantly by the client you have running locally, as well as the subscribed terminal instance.
+
+To publish a message in your code, you can add the following line to your @main@ method, above the line @// Keep the program running@, after subscribing to the channel:
+
+```[java]
+channel.publish("", "A message sent from my first client!");
```

Comment:
don't like empty strings, I think it's better to use some text for name: "message" or "update"